src/include/network/postgres/postgres_protocol_interpreter.h

```diff
@@ -157,12 +157,30 @@ class PostgresProtocolInterpreter : public ProtocolInterpreter {
     return nullptr;
   }
 
+  /**
+   * @param fingerprint key
+   * @param statement statement to take ownership of
+   */
+  void AddStatementToCache(const std::string &fingerprint, std::unique_ptr<network::Statement> &&statement) {
+    statement_cache_[fingerprint] = std::move(statement);
+  }
+
+  /**
+   * @param fingerprint key
+   * @return Statement if it exists in the cache, otherwise nullptr
+   */
+  common::ManagedPointer<network::Statement> LookupStatementInCache(const std::string &fingerprint) const {
```

Comment:
Shouldn't there be a way to remove something from the cache as well to account for things like invalidation or eviction?

Reply:
We haven't defined what the invalidation policies are yet (besides a DDL change, but we don't know how to orchestrate that yet). I can add a placeholder API if it makes sense, but I'm not sure we should without thinking through all of the safeguards and constraints.